### DISCUSSION TITLE
selected items for togglebuttons

### DISCUSSION THREAD

**Discussion by phpust** [filamentphp/filament#12210] [Help]
### Package

Form builder

### Package Version

v3.2

### How can we help you?

how to set selected list for some of ToggleButtons options.
something like this
```
Forms\Components\ToggleButtons::make('words.list')
    ->options([
        'draft' => 'Draft',
        'scheduled' => 'Scheduled',
        'published' => 'Published'
    ])
    ->selected(['draft','published'])
    ->inline(),`

**Comment by leandrocfe** [comment]
```php
->multiple()
->default(['draft', 'published'])
```
?

**Reply by phpust** [reply]
this is not valid. when i activate multiple, default is not working at all, also in multiple mode when i select one button, all of them will be selected together

**Reply by leandrocfe** [reply]
Are you using the panel builder or only the form builder?

**Reply by micraux** [reply]
The `Toggle` component is made for only boolean values (true/false), so you can't have 3 options for 1 toggle.

For your use case, you have to create 3 toggles, something like:
```php
Toggle::make('draft')
    ->default(true),

Toggle::make('scheduled')
    ->default(false),

Toggle::make('published')
    ->default(true),
```

As an alternative, you may want to have a look at:
`CheckboxList`: [https://filamentphp.com/docs/3.x/forms/fields/checkbox-list](https://filamentphp.com/docs/3.x/forms/fields/checkbox-list)
or
`Select` with `->multiple()` as suggested by @leandrocfe : [https://filamentphp.com/docs/3.x/forms/fields/select#multi-select](https://filamentphp.com/docs/3.x/forms/fields/select#multi-select)

**Reply by leandrocfe** [reply]
ToogleButtons, actually
https://filamentphp.com/docs/3.x/forms/fields/toggle-buttons#selecting-multiple-buttons
